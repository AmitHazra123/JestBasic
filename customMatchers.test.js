test('should be match the cases', () => {
  const mockFunc = jest
    .fn()
    .mockReturnValue('default')
    .mockImplementationOnce((err, val) => true)
    .mockImplementationOnce((err, val1, val2) => false)
    .mockName('a mock name');

  mockFunc(null, 5);
  mockFunc(null, 7, 9);
  
  // The mock function was called at least once
  expect(mockFunc).toHaveBeenCalled();

  // The mock function was called at least once with the specified args
  expect(mockFunc).toHaveBeenCalledWith(null, 5);

  // The last call to the mock function was called with the specified args
  expect(mockFunc).toHaveBeenLastCalledWith(null, 7, 9);

  // All calls and the name of the mock is written as a snapshot
  expect(mockFunc).toMatchSnapshot();


  // The mock function was called at least once
  expect(mockFunc.mock.calls.length).toBeGreaterThan(0);

  // The mock function was called at least once with the specified args
  expect(mockFunc.mock.calls).toContainEqual([null, 5]);

  // The last call to the mock function was called with the specified args
  expect(mockFunc.mock.calls[mockFunc.mock.calls.length - 1]).toEqual([
    null,
    7,
    9
  ]);

  // The first arg of the last call to the mock function was `42`
  // (note that there is no sugar helper for this specific of an assertion)
  expect(mockFunc.mock.calls[mockFunc.mock.calls.length - 1][0]).toBe(null);

  // A snapshot will check that a mock was invoked the same number of times,
  // in the same order, with the same arguments. It will also assert on the name.
  expect(mockFunc.mock.calls).toEqual([[null, 5], [null, 7, 9]]);
  expect(mockFunc.getMockName()).toBe('a mock name');
});