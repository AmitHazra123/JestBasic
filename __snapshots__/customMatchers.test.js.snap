// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be match the cases 1`] = `
[MockFunction a mock name] {
  "calls": Array [
    Array [
      null,
      5,
    ],
    Array [
      null,
      7,
      9,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": true,
    },
    Object {
      "type": "return",
      "value": false,
    },
  ],
}
`;
